app/services/github_issues_author.rb

```diff
@@ -0,0 +1,48 @@
+class GithubIssuesAuthor
+  ALLOWED_ACTIONS = %w[opened edited deleted].freeze
+
+  def perform(action, issue)
+    return unless ALLOWED_ACTIONS.include?(action) && valid_issue?(issue)
+
+    send("issue_#{action}", issue)
+  end
+
+  private
+
+  def valid_issue?(issue)
+    %w[id title body].all? { |attribute| issue.key? attribute } &&
+      issue['pull_request'].nil?
+  end
+
+  def issue_opened(issue)
+    ActiveRecord::Base.transaction do
+      Book.create!(title:     Faker::Book.title,
```

Comment:
I always prefer the failure to be load, but i'm not sure if i should be rescuing from InvalidRecord here or not, what do you think ?

Reply:
Yes, sorry typo 👍

Reply:
yeah, agree with that
if there's some unexpected failure here we want to know about it